internal/client/consensus/grandpa/communication.go

```diff
@@ -0,0 +1,849 @@
+// Copyright 2025 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package grandpa
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/ChainSafe/gossamer/internal/client/keystore"
+	"github.com/ChainSafe/gossamer/internal/client/network"
+	gossip "github.com/ChainSafe/gossamer/internal/client/network-gossip"
+	"github.com/ChainSafe/gossamer/internal/client/network/service"
+	networkSync "github.com/ChainSafe/gossamer/internal/client/network/sync"
+	peerid "github.com/ChainSafe/gossamer/internal/client/network/types/peer-id"
+	primitives "github.com/ChainSafe/gossamer/internal/primitives/consensus/grandpa"
+	"github.com/ChainSafe/gossamer/internal/primitives/runtime"
+	grandpa "github.com/ChainSafe/gossamer/pkg/finality-grandpa"
+	"github.com/ChainSafe/gossamer/pkg/scale"
+)
+
+// How often to rebroadcast neighbor packets, in cases where no new packets are created.
+const neighborRebroadcastPeriod = 2 * 60 * time.Second
+
+// cost scalars for reporting peers.
```

Comment:
Nit/ could way maybe describe the cost (here and below) as something besides scalars? I see what it means, but it is a bit confusing to me. Maybe something like cost value or cost amount, or or just simply cosy

Reply:
Revised.